src/kafka_clj_utils/producers.clj

```diff
@@ -21,16 +21,20 @@
 
 (s/def ::eventId ::ku/non-blank-string)
 (s/def ::metadata (s/keys :req-un [::eventId]))
-(s/def ::record (s/keys :req-un [::metadata]))
+;; metadata is optional now as it's possible to override with the key-fn and use a different field
```

Comment:
I'm not 100% sure about this.
While metadata isn't needed if the user provides a `key-fn`, I think it having a `:req-un [::eventId]` will force a particular style of metadata. Give metadata is vague and isn't namespace it might catch someone of guard.
Is it worth changing that too? Or is there a better spec to capture working with the key-fn